### ISSUE TITLE
Recursion Error in Iceweasel 49, trying to run the gopherjs/jquery example in README.md

### ISSUE THREAD

**Issue by ZenifiedFromI2P** [gopherjs/gopherjs#546]
I tried to run the jquery example in gopherjs/jquery, in my Iceweasel 49 (Firefox), note I have NoScript and few privacy addons enabled (uBlock Origin, HTTPS Everywhere, etc.)..

```
Your current jQuery version is: 2.1.0
too much recursion <Learn More> main.js:int:int
```
go env (It may be useless):
```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/user/gopath"
GORACE=""
GOROOT="/usr/lib/go"
GOTOOLDIR="/usr/lib/go/pkg/tool/linux_amd64"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build875613271=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
```
Edit: Found out more after debugging:

It issues at main.js:2059:1, main.js is here: https://bpaste.net/raw/9652fed5cdcc

**Comment by shurcooL** [comment]
I've tried your .js with jQuery version 2.2.4 (from jQuery 2.x section at https://code.jquery.com/) in latest stable version of Chrome (54.0.2840.87) on macOS 10.12.1.

It works without issue here. All I see in console is:

![image](https://cloud.githubusercontent.com/assets/1924134/20034507/134dee34-a37d-11e6-9add-81bbbf24a187.png)

I also tried it in Firefox Developer Edition (51.0a2) that I happen to have installed, and it has warnings about unreachable code, but no errors:

![image](https://cloud.githubusercontent.com/assets/1924134/20034511/874e5b5c-a37d-11e6-8b1a-79cffcab11b2.png)

Is Iceweasel 49 the latest stable version of that browser? When was it released? What browser is it? I'm not very familiar with it. Is it https://en.wikipedia.org/wiki/GNU_IceCat? If so, it says the stable release of that is 38.8.0, so how do you have 49?

In general, GopherJS supports only current versions of modern evergreen browsers, so I'm not sure if Iceweasel is meant to be supported or not.

**Comment by ZenifiedFromI2P** [comment]
Debian's Firefox fork is named Iceweasel, I have the latest from Parabola repos..

**Comment by ZenifiedFromI2P** [comment]
Well, also can you try typing in text in the input? It prints errors in the console instead of editing the span (callback), that's where the recursion error occurs.

**Comment by shurcooL** [comment]
> Debian's Firefox fork is named Iceweasel, I have the latest from Parabola repos..

I see, thanks.

> Well, also can you try typing in text in the input?

I'd like to, but I didn't see any input. Is it a part of the HTML? Can you share that?

**Comment by ZenifiedFromI2P** [comment]
I reinstalled my OS, now I have: Firefox 47.0 (and 50 soon)..

I tried again...

Do you see anything in the console while trying any input? it errors as recursion error in my case...

**Comment by ZenifiedFromI2P** [comment]
The sources was meant to SetHTML on the span#output element the text you enter on the input...

**Comment by ZenifiedFromI2P** [comment]
HTML code:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>Welcome to GopherJS with jQuery</title>
    <link rel="stylesheet" href="foundation.min.css" />
	<style>
	body {
		padding-left: 10%;
		padding-right: 10%;
		padding-top: 2%;
		padding-bottom: 2%;
  }
	</style>
</head>
<body>
    <input id="name" type="text" value="" placeholder="What is your name ?" autofocus/>
    <span id="output"></span>
    <script src="jquery.js"></script>
    <script src="main.js"></script>
</body>
</html>
```

**Comment by ZenifiedFromI2P** [comment]
Also, when I try to run code on Playground, it works, but when I run it locally it fails with error like 'v.$get' not a function..

**Comment by ZenifiedFromI2P** [comment]
```go
package main
import (
  "github.com/gopherjs/gopherjs/js"
  "golang.org/x/crypto/ed25519"
)
func main()  {
  js.Global.Set("ThisTestFails", map[string]interface{}{
  "New": New,
  })
}

func New() *js.Object {
  return js.MakeWrapper(&Antisocial{})
}

type Testing struct {
  EdPub []byte
  EdPriv []byte
  CurvePub []byte
  CurvePriv []byte
}

func (self *Testing) KeyGen() (pub, priv []byte){
  pub, priv, err := ed25519.GenerateKey(nil)
  if err != nil {
    panic(err)
  }
  self.EdPub = pub
  self.EdPriv = priv
  return pub, priv
}```

**Comment by ZenifiedFromI2P** [comment]
I build the above code, include it with <script> and open my console, type t = Testing.New()

It fails with 'v.$get' not found

**Comment by ZenifiedFromI2P** [comment]
@shurcooL any updates?

**Comment by shurcooL** [comment]
Thanks for posting the HTML, I am able to reproduce with that. When I type some text in the input box, I get "Uncaught RangeError: Maximum call stack size exceeded" errors:

![image](https://cloud.githubusercontent.com/assets/1924134/20250118/2f868f10-a9be-11e6-8a25-12046a842ad8.png)

It looks like a programming error to me. There is infinite recursion:

![image](https://cloud.githubusercontent.com/assets/1924134/20250121/4a1cca06-a9be-11e6-96c0-7cdad80fae92.png)

**Comment by shurcooL** [comment]
I tried the `ThisTestFails` snippet you posted, and not seeing an issue there.

![image](https://cloud.githubusercontent.com/assets/1924134/20250130/b460064e-a9be-11e6-9bcf-5782f282ab89.png)

Can you verify you have the latest version of GopherJS? Try updating it with `go get -u -v github.com/gopherjs/gopherjs`. If you have latest version, that command should not print anything.

**Comment by ZenifiedFromI2P** [comment]
OK I updated my GopherJS, I will try it again..

**Comment by ZenifiedFromI2P** [comment]
```sh
[user@localhost ~]$ gopherjs version
GopherJS 1.7-1
```

**Comment by ZenifiedFromI2P** [comment]
Any updates when the JQ problem above will be fixed?

**Comment by shurcooL** [comment]
> OK I updated my GopherJS, I will try it again..

Ok. What was the result, is the `ThisTestFails` issue still there?

> Any updates when the JQ problem above will be fixed?

Which problem are you referring to?

The current status is that I can't reproduce the `ThisTestFails` issue, and "Uncaught RangeError: Maximum call stack size exceeded" looks like a user programming error, not a problem in GopherJS to me.

**Comment by ZenifiedFromI2P** [comment]
No, the first error was a example..

The JQ error is not a user error, it happens with JS tries to switch on
the type of the object (event) and tries to assert it's the correct
type... This results in a infinite loop...

Will try again..

**Comment by ZenifiedFromI2P** [comment]
No, the first error was a example..

The JQ error is not a user error, it happens with JS tries to switch on
the type of the object (event) and tries to assert it's the correct
type... This results in a infinite loop...

Will try again..

**Comment by shurcooL** [comment]
> The JQ error is not a user error, it happens with JS tries to switch on
the type of the object (event) and tries to assert it's the correct
type... This results in a infinite loop...

Can you post the Go code that reproduces that issue?

**Comment by ZenifiedFromI2P** [comment]
The code in gopherjs/jquery 's README.md

**Comment by ZenifiedFromI2P** [comment]
Any updates?

**Comment by ZenifiedFromI2P** [comment]
```go
package main

import "github.com/gopherjs/jquery"

//convenience:
var jQuery = jquery.NewJQuery

const (
    INPUT  = "input#name"
    OUTPUT = "span#output"
)

func main() {

    //show jQuery Version on console:
    print("Your current jQuery version is: " + jQuery().Jquery)

    //catch keyup events on input#name element:
    jQuery(INPUT).On(jquery.KEYUP, func(e jquery.Event) {

        name := jQuery(e.Target).Val()
        name = jquery.Trim(name)

        //show welcome message:
        if len(name) > 0 {
            jQuery(OUTPUT).SetText("Welcome to GopherJS, " + name + " !")
        } else {
            jQuery(OUTPUT).Empty()
        }
    })
}
```

**Comment by ZenifiedFromI2P** [comment]
OK, I updated my gopherjs and that pet example now works...

**Comment by ZenifiedFromI2P** [comment]
GopherJS source directory hasn't seen a commit in 2 months, is this dead? (I see three unsolved 'bugs' in the issues)

**Comment by shurcooL** [comment]
> Any updates?

Sorry, I haven't had the time to take another look at this yet.

> OK, I updated my gopherjs and that pet example now works...

Great. You should be using the latest version of GopherJS, anything older may have issues that have already been resolved.

> GopherJS source directory hasn't seen a commit in 2 months, is this dead? (I see three unsolved 'bugs' in the issues)

The development of this project is not as active as it might've been earlier, but I wouldn't call it dead. Most of the remaining bugs are relatively minor and harder to fix, so it takes longer.

**Comment by ZenifiedFromI2P** [comment]
I may have spotted some issues.

Firstly, GopherJS seems to emit the 'v.$get is not a function' when the packages are vendored.

Secondly, I have spotted weird errors, like I first time run the compiler, it fails, (and without changing the source or anything in the directory), I restart it after deleting *.js *.map, it works..

**Comment by shurcooL** [comment]
> when the packages are vendored.

Which packages, the ones you're importing? Or GopherJS itself? Using a vendored copy of GopherJS is not supported, see https://github.com/gopherjs/gopherjs#what-is-supported. Vendored packages in your code should be okay, but I don't have much experience with that so I can't be sure.

> Secondly, I have spotted weird errors, like I first time run the compiler, it fails, (and without changing the source or anything in the directory), I restart it after deleting *.js *.map, it works..

Hmm, I haven't seen that before. If you can provide clear reproducible steps to trigger that, please file a new issue and describe it there. This issue is getting pretty long and unfocused, so it's hard to do anything specific here.

**Comment by ZenifiedFromI2P** [comment]
The packages I'm importing..

**Comment by shurcooL** [comment]
If I understand you correctly, it could be a valid bug then, so it'd be helpful to have an issue tracking it if we don't already. Doing so in this issue isn't a good idea, it's very long winded and hard to follow.

If you're interested in reporting an issue, I suggest you read https://github.com/primer/primer-markdown/blob/master/.github/CONTRIBUTING.md#bug-reports to learn about what makes a good bug report. Thanks!

I'll close this issue because there's nothing actionable we can do in GopherJS to resolve this.
